server/src/main/java/org/apache/druid/server/coordinator/compact/CompactionStatus.java

```diff
@@ -305,17 +305,30 @@ private CompactionStatus metricsSpecIsUpToDate()
       if (ArrayUtils.isEmpty(configuredMetricsSpec)) {
         return COMPLETE;
       }
+      final AggregatorFactory[] configuredMetricsCombiningFactorySpec =
```

Comment:
Why are we comparing just the combining factory now and not the original AggregatorFactory itself? Please add comments (and tests, if not already added) to validate this change.

Reply:
Added it as a javadoc for the function

Reply:
I would advise retaining the old version of the code and doing this change in a follow up. In the current PR, I would prefer if we didn't modify any of the logic for native compaction. What happens if we just stick to the old logic?

Reply:
With the last change to unsupport problematic aggregator factory definitions, this change is not required. Have reverted it.